layers/layers.go

```diff
@@ -0,0 +1,273 @@
+package layers
+
+import (
+	"errors"
+	"io"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/docker/daemon/graphdriver"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/stringid"
+)
+
+var (
+	ErrLayerDoesNotExist = errors.New("layer does not exist")
+)
+
+type LayerAddress digest.Digest
+
+type TarStreamer interface {
+	TarStream() (io.Reader, error)
+}
+
+// Layer represents a read only layer
+type Layer interface {
+	TarStreamer
+	Address() LayerAddress
+	Parent() (Layer, error)
+	Size() (int64, error)
```

Comment:
Why interface and why not just properties of a defined struct(parent would be Parent digest).